test/jdk/java/awt/color/ICC_Profile/BuiltInProfileCheck.java

```diff
@@ -0,0 +1,124 @@
+    private static final String EXCEPTION_MSG = "Built-in profile cannot be modified";
+
+    private static final Map<Integer, String> colorSpace = Map.of(
+            ColorSpace.CS_sRGB, "CS_sRGB",
+            ColorSpace.CS_PYCC, "CS_PYCC",
+            ColorSpace.CS_GRAY, "CS_GRAY",
+            ColorSpace.CS_CIEXYZ, "CS_CIEXYZ",
+            ColorSpace.CS_LINEAR_RGB, "CS_LINEAR_RGB"
+    );
+
+    public static void main(String[] args) throws IOException {
+        System.out.println("CASE 1: Testing BuiltIn Profile");
+        for (int cs : colorSpace.keySet()) {
+            updateProfile(true, cs);
+        }
+        System.out.println("Passed\n");
+
+        System.out.println("CASE 2: Testing Custom Profile");
+        updateProfile(false, ColorSpace.CS_sRGB);
+        System.out.println("Passed\n");
+
+        System.out.println("CASE 4: Testing Built-In Profile"
+                           + " Serialization & Deserialization");
+        testSerialization(true);
+        System.out.println("Passed\n");
+
+        System.out.println("CASE 5: Testing Custom Profile"
+                           + " Serialization & Deserialization");
+        testSerialization(false);
+        System.out.println("Passed\n");
+    }
+
+    private static void updateProfile(boolean isBuiltIn, int cs) {
+        ICC_Profile builtInProfile = ICC_Profile.getInstance(cs);
+        // if isBuiltIn=true use builtInProfile else create a copy
+        ICC_Profile iccProfile = isBuiltIn ? builtInProfile
+                                 : ICC_Profile.getInstance(builtInProfile.getData());
+
+        byte[] headerData = iccProfile.getData(HEADER_TAG);
+        // Set profile class to valid icSigInputClass = 0x73636E72
+        headerData[INDEX] = 0x73;
+        headerData[INDEX + 1] = 0x63;
+        headerData[INDEX + 2] = 0x6E;
+        headerData[INDEX + 3] = 0x72;
+
+        if (isBuiltIn) {
+            System.out.println("Testing: " + colorSpace.get(cs));
+            try {
+                // Try updating a built-in profile, IAE is expected
+                iccProfile.setData(HEADER_TAG, headerData);
+                throw new RuntimeException("Test Failed! IAE NOT thrown for profile"
+                                           + colorSpace.get(cs));
+            } catch (IllegalArgumentException iae) {
+                if (!iae.getMessage().equals(EXCEPTION_MSG)) {
+                    throw new RuntimeException("Test Failed! IAE with exception msg \""
+                                               + EXCEPTION_MSG + "\" NOT thrown for profile "
+                                               + colorSpace.get(cs));
+                }
+            }
+        } else {
+            // Modifying custom profile should NOT throw IAE
+            iccProfile.setData(HEADER_TAG, headerData);
+        }
+    }
+
+    // this test is added to check the ICC_Profile's transient isBuiltIn flag
+    private static void testSerialization(boolean isBuiltIn)
```

Comment:
This method does not actually test serialization; instead, it clones/copies the ICC profile via the filesystem. This is why you did not get an exception at the end of this method, even for a built-in profile.

You should use something like this instead:

```java
ByteArrayOutputStream baos = new ByteArrayOutputStream();
ObjectOutputStream oos = new ObjectOutputStream(baos);
oos.writeObject(iccProfile);
```
```java
byte[] array = baos.toByteArray();
ObjectInputStream ois = new ObjectInputStream(new ByteArrayInputStream(array));
ICC_Profile iccProfile = (ICC_Profile) ois.readObject();
```
In the example above, the built-in profile should throw an IAE, but a custom profile should not. (I suggest covering all built-in profiles as well).